### ISSUE TITLE
ERROR in ./node_modules/prettier/index.js

### ISSUE THREAD

**Issue by GGICE** [prettier/prettier#3925]
In my project (webpack bebal  node), use prettier 1.10.2, use prettier 1.9.0 no this problem.
use  `const prettier = require("prettier");`  will output this error.

[1] ERROR in ./node_modules/prettier/index.js
[1] Module parse failed: Invalid labeled declaration (4071:8)
[1] You may need an appropriate loader to handle this file type.
[1] |     }
[1] |     _token_stack:
[1] |         var lex = function () {
[1] |             var token;
[1] |             token = lexer.lex() || EOF;
[1]  @ dll renderer
[1] npm ERR! code ELIFECYCLE

**Comment by lydell** [comment]
Hi! Sad to hear you are having problems after upgrading Prettier.

It’s really hard to say what’s going on here. Could you make a minimal test project that demonstrates your problem? So we can run:

1. `git clone https://github.com/GGICE/your-minimal-test-project-demo.git`
2. `npm install`
3. `npm test`

… to reproduce and debug your issue.

**Comment by GGICE** [comment]
In project folder only  `index.js` and `webpack.config.js`, run cmd `node index.js` is ok, but use run webpcak cmd  `webpack`  will throw error, build fail.

```
ERROR in ./node_modules/prettier/index.js
Module parse failed: Invalid labeled declaration (4071:8)
You may need an appropriate loader to handle this file type.
|     }
|     _token_stack:
|         var lex = function () {
|             var token;
|             token = lexer.lex() || EOF;
 @ ./index.js 1:17-36`
```
index.js

```
const prettier = require("prettier");
const result = prettier.formatWithCursor(`vue`, {
  parser: 'vue'
});
console.log(result);
```

webpack.config.js

```
const path = require('path');

module.exports = {
  entry: "./index.js",

  output: {
    path: path.resolve(__dirname, 'dist/'),
    filename: '[name].js',
  },

  module: {
    rules: []
  }
}
```

**Comment by lydell** [comment]
Thanks!

webpack needs to parse all JS files to find all `require()` calls. It parses using [Acorn](https://github.com/acornjs/acorn).

node_modules/prettier/index.js contains both `'use strict'` and a labeled `var` statement `_token_stack: var lex = …`.

Acorn disallowed that in https://github.com/acornjs/acorn/issues/393

But then allowed it again in https://github.com/acornjs/acorn/issues/521

The Acorn that webpack uses comes from [acorn-dynamic-import](https://github.com/kesne/acorn-dynamic-import). However, webpack uses v2 of that module, which depends on v4 of Acorn, which does not allow those labeled `var` statements. Luckily, v3 has updated to v5 of Acorn which _does_ allow them again.

This webpack PR updates to v3 of acorn-dynamic-import: https://github.com/webpack/webpack/pull/6254. It was merged into the `next` branch of webpack. So you’ll either have to wait for that to be released or use the beta version of webpack or something.

Another thing you can do is to add code to your build step that removes the `'use strict'` line at the top of node_modules/prettier/index.js.

I think there’s nothing we can do on Prettier’s side here, so I’m closing this issue. Feel free to keep commenting here, though! We’ll reopen if I’m wrong.

**Comment by GGICE** [comment]
Thank you help ！
